src/main.zig

```diff
@@ -2177,9 +2177,19 @@ fn buildOutputType(
                 const sdk = std.zig.system.darwin.getDarwinSDK(arena, target_info.target) orelse
                     break :outer false;
                 native_darwin_sdk = sdk;
-                try clang_argv.ensureUnusedCapacity(2);
+
+                try clang_argv.ensureUnusedCapacity(4);
                 clang_argv.appendAssumeCapacity("-isysroot");
                 clang_argv.appendAssumeCapacity(sdk.path);
+
+                // NOTE: it seems like providing the sysroot isn't enough
+                // neither --sysroot or -isysroot allow clang to find the sdk libc headers,
+                // so we manually add them here
+
+                clang_argv.appendAssumeCapacity("-isystem");
```

Comment:
Hmm, I don't think this should be needed. The purpose of the sysroot is to prefix every absolute include dir with the said sysroot path. What you are doing here is adding an explicit `-isystem` path that circumvents the sysroot path altogether and explicitly includes the concatenation of `sysroot` with `usr/include`. I do not believe this is the right solution to the problem. Instead, we should do more investigation what paths are included and work out why clang cannot find the headers.